src/cc/perf_reader.c

```diff
@@ -172,6 +172,21 @@ void perf_reader_event_read(struct perf_reader *reader) {
   // Consume all the events on this ring, calling the cb function for each one.
   // The message may fall on the ring boundary, in which case copy the message
   // into a malloced buffer.
+  // 
+  // NOTE: The loop condition is different from the official BCC repo. The official impl code is
+  // as follows:
+  // for (data_head = read_data_head(perf_header); perf_header->data_tail != data_head;
+  //      data_head = read_data_head(perf_header)) {
+  //     ... Loop body
+  // }
+  //
+  // As stated in https://github.com/iovisor/bcc/pull/3822#issuecomment-1023775038,
+  // the official BCC code above can handle occasional burst by eagerly depleting the perf buffer.
+  //
+  // The worst case scenario is that the producer constantly produces more data
+  // than the consumer can keep up, and results into an infinite loop. Pixie team have observed
+  // a few cases where an perf buffer polling iteration took very long time, which was fixed
+  // by this new loop condition.
```

Comment:
...than the consumer can process, and results in an infinite loop. The Pixie team has observed
a few cases where a perf buffer polling iteration took a very long time, which was fixed
by this new loop condition.